designs/helm-chart-autogenerate-plugin.md

```diff
@@ -0,0 +1,287 @@
+crd:
+  ## Whether the `resources` field contains `crd` in the `config/default/kustomization.yaml` file
+  create: false
+
+## RBAC configuration under the `config/rbac` directory
+rbac:
+  ## Whether the `resources` field contains `rbac` in the `config/default/kustomization.yaml` file
+  create: true
+  serviceAccountName: "controller-manager"
+
+
+## Manager configuration referent to `config/manager/manager.yaml`
+manager:
+  ## The Security Context of the manager Pod.
+  ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
+  podSecurityContext:
+    runAsNonRoot: true
+
+  ## Configure the kubeRbacProxy container.
+  kubeRbacProxy:
+    image:
+     registry: gcr.io
+     repository: "kubebuilder/kube-rbac-proxy"
+     tag: v0.13.0
+     pullPolicy: IfNotPresent
+
+  ## Configure the resources.
+  ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+  resources:
+    limits:
+      cpu: 500m
+      memory: 128Mi
+    requests:
+      cpu: 5m
+      memory: 64Mi
+
+  ## Configure the controller container.
+  controller:
+    image:
+      registry: docker.io
+      repository: ""
+      tag: latest
+      pullPolicy: IfNotPresent
+
+    ## Configure extra options for liveness probe.
+    ## More info: https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-probes/#configure-probes
+    livenessProbe:
+      enabled: true
+      httpGet:
+        path: /healthz
+        port: 8081
+      initialDelaySeconds: 15
+      periodSeconds: 20
+
+    readinessProbe:
+      enabled: true
+      httpGet:
+        path: /readyz
+        port: 8081
+      initialDelaySeconds: 5
+      periodSeconds: 10
+
+    ## Configure the resources.
+    ## More info: https://kubernetes.io/docs/concepts/configuration/manage-resources-containers/ 
+    resources:
+      limits:
+        cpu: 500m
+        memory: 128Mi
+      requests:
+        cpu: 10m
+        memory: 64Mi
+
+     ## Configure the rbac roles.
+     serviceAccountName: ""
+
+## Webhook configuration under the `config/webhook` directory
+webhook:
+  ## Whether the `resources` field contains `webhook` in the `config/default/kustomization.yaml` file
+  enabled: true
+  ## More info: https://kubernetes.io/docs/concepts/services-networking/service/
+  service:
+    ports:
+    - name: webhook-server
+      protocol: TCP
+      containerPort: 9443
+
+metrics:
+  enabled: true
+  service:
+    type: ClusterIP
+    ports:
+    - port: 8443
+      protocol: TCP
+
+## Configure the cert-manager dependency.
+certmanager:
+  enabled: true
+  installCRDs: true
+  ## Add the owner reference to the certificate.
+  extraArgs:
+  - --enable-certificate-owner-ref=true
+
+## Prometheus configuration under the `config/prometheus` directory
+prometheus:
+  ## Whether the `resources` field contains `prometheus` in the `config/default/kustomization.yaml` file
+  enabled: false
```

Comment:
```suggestion
```

We cannot manage the dependencies.
We do not know where those will be installed.
For example, if a solution is installed in OCP it has Prometheus already.